### ISSUE TITLE
How to improve the value of map0.5 and jump out of local optimum?

### ISSUE THREAD

**Issue by Mrgao9** [ultralytics/yolov5#1985]
## ❔Question
Training custom data set, I only train one class, there are 37943 ship sample images in the training set and 4742 ship sample images in the verification set；

After 70 rounds, the value of map0.5 has been locally optimal, and has not improved since then. What is a good way to get out of local optimum?
The maximum value of map0.5 was 0.713 in the 70th round, and then began to decline;

My pre training model is yolov5s.pt。

Looking forward to your reply! Thank you very much.

The training log is as follows：
![image](https://user-images.githubusercontent.com/22071198/105120768-d0c1db00-5b0d-11eb-875c-82d211f04be3.png)

![image](https://user-images.githubusercontent.com/22071198/105120829-ed5e1300-5b0d-11eb-8aed-1643dab642c7.png)
![image](https://user-images.githubusercontent.com/22071198/105120871-0666c400-5b0e-11eb-90fd-828e15228678.png)

**Comment by glenn-jocher** [comment]
@Mrgao9 improving results is very simple:

- Use a larger dataset
- Use a larger model i.e. `--weights yolov5x.pt`
- Use a larger image size i.e. `--img 1280`
- Train longer i.e. `--epochs 500`
- Evolve hyperparameters (see Tutorials) https://docs.ultralytics.com/yolov5

**Comment by Mrgao9** [comment]
> @Mrgao9 improving results is very simple:
> 
> * Use a larger dataset
> * Use a larger model i.e. `--weights yolov5x.pt`
> * Use a larger image size i.e. `--img 1280`
> * Train longer i.e. `--epochs 500`
> * Evolve hyperparameters (see Tutorials) https://docs.ultralytics.com/yolov5

First of all, thank you very much for your reply!  I will try your good advice!
 Is it a way to jump out of local optimum to use "CosineAnnealingLR" or "OneCycleLR" learning rate adjustment strategies?

**Comment by glenn-jocher** [comment]
@Mrgao9 we already employ OneCycleLR by default, you might try varying the LR scheduler of course. You may want to examine your individual loss components, and if one is overfitting more than the others, you might reduce it's contribution to the loss function by reducing it's associated hyperparameter, i.e. hyp['obj'] if objectness is overfitting most etc.

**Comment by Mrgao9** [comment]
> @Mrgao9 we already employ OneCycleLR by default, you might try varying the LR scheduler of course. You may want to examine your individual loss components, and if one is overfitting more than the others, you might reduce it's contribution to the loss function by reducing it's associated hyperparameter, i.e. hyp['obj'] if objectness is overfitting most etc.

![image](https://user-images.githubusercontent.com/22071198/105272636-bdc60e00-5bd4-11eb-8cfe-30bc7191820c.png)
![image](https://user-images.githubusercontent.com/22071198/105272866-2b723a00-5bd5-11eb-8772-702803beeac3.png)

You use the cosine annealing principle in function one_cycle，this is the "OneCycleLR" learning rate adjustment strategies ?

**Comment by glenn-jocher** [comment]
@Mrgao9 onecyclelr runs a cosine LR for one cycle, i.e. a half wavelength cosine. You can see the LRs in your W&B or tensorboard logs:

<img width="1704" alt="Screen Shot 2021-01-20 at 7 28 20 PM" src="https://user-images.githubusercontent.com/26833433/105276196-abf15600-5b55-11eb-8218-b843d9f5a8fd.png">

**Comment by Mrgao9** [comment]
> @Mrgao9 onecyclelr runs a cosine LR for one cycle, i.e. a half wavelength cosine. You can see the LRs in your W&B or tensorboard logs:
> 
> <img alt="Screen Shot 2021-01-20 at 7 28 20 PM" width="1704" src="https://user-images.githubusercontent.com/26833433/105276196-abf15600-5b55-11eb-8218-b843d9f5a8fd.png">

I understand your explanation. Thank you very much for your help. I wish you a happy life！

**Comment by Mrgao9** [comment]
> @Mrgao9 improving results is very simple:
> 
> * Use a larger dataset
> * Use a larger model i.e. `--weights yolov5x.pt`
> * Use a larger image size i.e. `--img 1280`
> * Train longer i.e. `--epochs 500`
> * Evolve hyperparameters (see Tutorials) https://docs.ultralytics.com/yolov5

@glenn-jocher Dear author, why can a larger image size improve the result?  If I use a larger image size to train, can I use yolov5s.pt for pre training model ?

**Comment by glenn-jocher** [comment]
@Mrgao9 that's an interesting question. I would say if you are training at less than native resolution then it's quite obvious why increasing resolution helps, but there are scenarios were training at larger-than-native resolution helps as well. COCO is a prime example here, where performance at --img 1280 is much better than perfromance at --img 640, despite the fact that no COCO image is larger than 640 pixels. 

From a statistical standpoint a 640-sized COCO image contains 100% of the available (Fisher) information the image will ever provide, upscaling it to 1280 is not creating additional information. The only conclusion I can draw from the empirical improvements observed is that the common CNN structures used today (YOLO, EfficientDet, etc) are not maximally efficient at native resolution, but have architectures instead tuned for efficiency at some larger multiple, perhaps between 2 and 4x native. 

Also note that supersampling at sub-pixel level becomes possible when upscaling an image, and this might play a part in the improvements (due to pixel discretization you can never convolve at sub-pixel levels at native resolution).

**Comment by glenn-jocher** [comment]
@Mrgao9 to answer your other question ideally you would start with a pretrained model trained at a similar --img resolution to what you want to train at, but failing that it can't hurt to use the default models which are trained at 640.

**Comment by Mrgao9** [comment]
@glenn-jocher Thank you very much for your detailed explanation. You are really nice! I will try your good advice!
